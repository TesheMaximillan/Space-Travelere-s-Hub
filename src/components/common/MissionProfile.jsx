import React from 'react';
import { useSelector } from 'react-redux';

function MissionProfile({ list }) {
  const missions = useSelector((state) => state.missionReducer);

  return (
    missions
    && missions.map(
      (mission) => mission.status && (
      <li key={mission.missionId} className={list}>
        {mission.missionName}
      </li>
      ),
    )
  );
}

export default MissionProfile;
