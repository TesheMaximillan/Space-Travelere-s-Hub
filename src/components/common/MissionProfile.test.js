import React from 'react';
import renderer from 'react-test-renderer';
import MissionProfile from './MissionProfile';
import renderContext from '../storeMock';

describe('MissionProfile component', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(renderContext(<MissionProfile />));
  });

  it('should render with given state from redux store', () => {
    // Assert
    expect(component.toJSON()).toMatchSnapshot();
  });
});
