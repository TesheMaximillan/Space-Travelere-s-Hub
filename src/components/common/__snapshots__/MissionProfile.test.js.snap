// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionProfile component should render with given state from redux store 1`] = `null`;
